File: src/_112/kotlin/Solution.kt
package _112.kotlin

import structure.TreeNode

/**
 * @author relish
 * @since 2018/05/02
 *
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int = 0) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 * [1] 1 true
 * [] 0 false
 */
class Solution {
    fun hasPathSum(root: TreeNode?, sum: Int): Boolean {
        if (root == null) return false
        if (root.left == null && root.right == null) return sum == root.`val`
        return hasPathSum(root.right, sum - root.`val`) || hasPathSum(root.left, sum - root.`val`)
    }
}

fun main(args: Array<String>) {
    println(Solution().hasPathSum(TreeNode.createTestData("[1]"), 1))
}

File: src/_112/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} sum
 * @return {boolean}
 */
var hasPathSum = function(root, sum) {
  if(root == null) {
      return false   
  }
  if(root.val === sum  && root.left === null && root.right === null) {
      return true
  }
  let sumNext = sum - root.val
  return hasPathSum(root.left, sumNext) || hasPathSum(root.right, sumNext)
};